edgePy/_DGEList.py

```diff
@@ -120,6 +120,10 @@ def samples(self, samples: Optional[np.ndarray]) -> None:
         # - Samples same length as ncol(self.counts) if defined
         if samples is not None:
             samples = np.array(list(self._format_fields(samples)))
+
+            if self.counts is not None:
+                if self.counts.shape[1] != len(samples):
+                    raise ValueError("Shape of counts does not match samples")
```

Comment:
It is best practice to raise any validation exception _before_ you do any work (the work appears on lines 121-122).